qemu/tests/block_resize.py

```diff
@@ -181,6 +182,30 @@ def get_md5_of_file(filename):
         session = vm.reboot(session=session)
 
         if os_type == 'linux':
+            # Add following code to solve the disk letter change issue on s390x
+            # virtio-blk, i.e. get new 'mpoint', 'partition' and 'disk'
+            machine_type = params['machine_type']
+            drive_format = params['drive_format']
+            if 's390' in machine_type and drive_format == 'virtio':
+                new_block_size = utils_disk.get_disk_size_linux(session, disk)
+                # Check if the disk letter changes by checking the disk size
+                if new_block_size != block_size:
```

Comment:
Is there a better way to check the disk? The block size is not the ID of the test disk, so sometimes it may be wrong.

The UUID of partition or slot of the disk is unique so you can get it once with the test disk, it should not be changed before/after reboot.
```
NAME         FSTYPE      LABEL UUID                                   MOUNTPOINT
sda
sda1       vfat              23E8-50BA                              /boot/efi
sda2       xfs               88250277-2091-4d4c-8a8d-eb9f4182805f   /boot
sda3       LVM2_member       kAkUVw-9FVW-63jw-UrAy-kDeU-QR70-OuNq9T
  rhel_dhcp158--102-root
            xfs               fe199486-953a-4e17-99be-98bae48de25b   /
  rhel_dhcp158--102-swap
             swap              56acdce6-2fc5-4bd4-82f5-2450e8ac975b   [SWAP]
sdb
sdb1       ext4              fb61022e-dce6-4e73-8f5c-59103f5ad339   /mnt/sdb1
```

Reply:
@PaulYuuu Updated to use the 'PTUUID' in comand 'blkid' output. Thanks.